File: python/block.py
"""

@author GalenS <galen.scovelL@gmail.com>
"""


class Block(dict):
    def __init__(self, index, timestamp, transactions, proof, previous_hash):
        self.index = index
        self.timestamp = timestamp
        self.transactions = transactions
        self.proof = proof
        self.previous_hash = previous_hash

        dict.__init__(
            self,
            index=index,
            timestamp=timestamp,
            transactions=transactions,
            proof=proof,
            previous_hash=previous_hash
        )

File: scala/Block.scala
package Backend

import java.util.Date


class Block(private val index: Int,
            private val timestamp: Date,
            private val transactions: Array[Transaction],
            private val proof: Int,
            private val previousHash: String) {


  def getIndex: Int = index

  def getTimestamp: Date = timestamp

  def getTransactions: Array[Transaction] = transactions

  def getProof: Int = proof

  def getPreviousHash: String = previousHash

  def getSortedJsonRepresentation: String = {
    s"{'index': '$index', 'proof': '$proof', 'previousHash': '$previousHash', 'timestamp': '$timestamp', 'transactions': '$transactions'"
  }
}
